ufl/algorithms/apply_function_pullbacks.py

```diff
@@ -64,10 +64,14 @@ def apply_single_function_pullbacks(g):
     mapping = element.mapping()
 
     r = ReferenceValue(g)
-
     gsh = g.ufl_shape
     rsh = r.ufl_shape
 
+    if element.mapping() == "physical":
+        # TODO: Is this right for immersed things
```

Comment:
This should be okay for k-surfaces in n-spaces, k<n, as far as the basis for your physically mapped elements is "scalar valued". AFAIK this is only an issue for tensor valued basis like Hdiv or Hdivdiv...

Reply:
It is unclear what the right thing is here.  Serendipity spaces are very fussy under mapping, and I have no idea if they can be defined on or mapped to an immersed manifold.

Reply:
I will update the comments to clarify this point.